### PR TITLE
Fix unit errors in ExternalCombiTimeTable constructor

### PR DESCRIPTION
**Pull request by henrikt-ma**
The errors are in relation to how the constructor is called in Modelica.Blocks.Sources.CombiTimeTable:
https://github.com/modelica/ModelicaStandardLibrary/blob/ce36c079036c3ade1c2b9f6626daf358ca7fb05f/Modelica/Blocks/Sources.mo#L1639-L1643

### CODE REVIEW

File: Modelica/Blocks/Types.mo [hunk 1]
```diff
@@ -107,11 +107,11 @@ package Types
       input String tableName "Table name";
       input String fileName "File name";
       input Real table[:, :];
-      input SI.Time startTime;
+      input Real startTimeScaled(unit = "1");
```
Comment (HansOlsson) [inline Modelica/Blocks/Types.mo:110]:
```suggestion
      input Real startTimeScaled;
```
I would prefer it without any unit.
The reason is that there might be users using the external object directly - without the time-scaling.

File: Modelica/Blocks/Types.mo [hunk 2]
```diff
@@ -107,11 +107,11 @@ package Types
...
       input Integer columns[:];
       input Modelica.Blocks.Types.Smoothness smoothness;
       input Modelica.Blocks.Types.Extrapolation extrapolation;
-      input SI.Time shiftTime=0.0;
+      input Real shiftTimeScaled(unit = "1")=0.0;
```
Comment (HansOlsson) [inline Modelica/Blocks/Types.mo:114]:
```suggestion
      input Real shiftTimeScaled=0.0;
```
And here as well.

Comment (henrikt-ma) [reply Modelica/Blocks/Types.mo:114]: Please motivate why to do it in a way that wouldn't let someone detect the mistake to believe that the table works with true (unscaled) time.

Comment (HansOlsson) [reply Modelica/Blocks/Types.mo:114]: But that is not a mistake - the table object works with what-ever you send in as time - as far as I recall the external object wasn't rewritten to handle scaled time. ( https://github.com/modelica/ModelicaStandardLibrary/commit/70f01e3f87338629f5cffc7e739785d8226ae178 - the C-code wasn't changed, and later correction was also outside the external object)

Note that this is only relevant if people directly use the external object. If they use the block (as normal people) then unit-errors will be detected at the block boundary.

Comment (henrikt-ma) [reply Modelica/Blocks/Types.mo:114]: Good, then we have a motivation.  Let's just leave this open a little longer to give more people time to react before the suggested change is applied.

**Review by beutlich — COMMENTED** [review]
Hm, in https://github.com/modelica/ModelicaStandardLibrary/issues/1473#issuecomment-272660281 I stated that the code is correct w.r.t. quantities and units.

**Comment by beutlich** [comment]
Changing the name of the ctor of the external object is a backwards-compatibility breaking change. I guess a conversion script will be required in that case.

**Comment by henrikt-ma** [comment]
> Changing the name of one input argument of the external object ctor is a backwards-compatibility breaking change. I guess a conversion script will be required in that case.

Right, but right now I don't think it looks like the argument names should be changed, since it apparently isn't the intention that those arguments must be "scaled time".

File: Modelica/Blocks/Types.mo (hunk 1 above)
Comment (henrikt-ma) [reply Modelica/Blocks/Types.mo]: The unit has now been removed.  Resolving.

File: Modelica/Blocks/Types.mo (hunk 2 above)
Comment (henrikt-ma) [reply Modelica/Blocks/Types.mo]: Enough time has passed, and @beutlich didn't seem to have any objections on this point.  The unit has now been removed.  Resolving.

**Review by HansOlsson — APPROVED** [review]
Looks good.

**Review by beutlich — APPROVED** [review]

**Pull request merged**